### ISSUE TITLE
Yacy for alternate TLDs

### ISSUE THREAD

**Issue by raisty** [yacy/yacy_search_server#231]
It is in roadmap list onion sites like AHMIA is listing it as well as alternate TLDs? 
https://github.com/juhanurmi/ahmia

Some alternative domains will be nice, if we speak about idea of decentralisation.
Example of TLDs: onion, eth (https://ens.domains/)

**Comment by luccioman** [comment]
Hi @raisty , up to now the idea for supporting alternate naming systems in YaCy was to write a new YaCy network definition file and configure the willing peers (at `/ConfigNetwork_p.html` page) to use it instead of the 'freeworld' network that is designed to support only standard domain names.

An example [network definition](https://github.com/yacy/yacy_search_server/blob/1ce38fdaedc097bd1027900d22e087dd704aeea4/defaults/yacy.network.zeronet.unit) is notably available for [ZeroNet](https://zeronet.io/) which operates on `.bit` domains.

**Comment by raisty** [comment]
As far as I know it is not enough definition, but gateway as well.
It should be implemented sofisticated system to browse alternative networks.
For onion TLD should be necessary some type of machine, which can handle all of the network.
Maybe it is connected to issue #232 as well.

**Comment by luccioman** [comment]
Yes you are completely right, YaCy network definition customization in itself is not sufficient, you also need to pass through some kind of gateway or alternate names resolver. You did not mention it, but there is notably the long running [OpenNIC](https://www.opennic.org/) project that manages its owns TLDs and also appears to be usable as a gateway to some peered naming systems.

By the way, as far as I know, in YaCy history it has been assumed that to keep things simple, support for non standard naming systems or protocols should be handled with external tools. Otherwise it would represent too much effort to integrate new libraries and code to support every new type of network appearing every year.

Maybe @Orbiter could complete on this, but I think that was roughly its explanation for how ZeroNet support should be added, and it still sounds reasonable to me.

**Comment by raisty** [comment]
Absolutely!
We need looking forward for decentralisation.
Great tool which can work is `lite nodes` - they will collect informations from different areas of network. Let's say it can be vertical line - from deep web to heaven web. Anybody can then fork lite nodes and deliver informations from craweling.
But in this case we need integration in YaCy and probably splitting crawelers in new project.
Contributing then can be simple.

Example of scheme:
```
- lite node - craweler populate YaCy instance
- node will store crawling results
- YaCy will connect to X amount of nodes
- YaCy will serve for frontend
```

OpenNIC is great example as well.
I am looking into anonymity and Blockchain structure.

Please comment, if you are open to discussion about decentralised googling. :godmode:
